### PR TITLE
Feature: Library Locking UI

### LINKED ISSUE #2994: "successfully updated" gets alerted after clicking settings buttons before any action is done
## Description

It can get a bit confusing when, after clicking a button in a project's settings, it will show an alert that says  "successfully updated" before the intended request is sent. For example, this happens with "Archive" and "Delete Project and Data" buttons. 

The notification is coming from a completed `dataInterface` call that triggers because it thinks clicking these buttons is submitting a form. Specifically starting from here: https://github.com/kobotoolbox/kpi/blob/5343e60cfa09e177dc48b4305bb5316c841a56bd/jsapp/js/components/modalForms/projectSettings.es6#L606

Can be seen on production through latest public collections

## Steps to Reproduce

1. Go to any created project's settings
2. Click "Delete Project and Data"
3. Don't delete and see alert

## Expected behavior 

Only confirm to user if DELETE request completed

## Actual behavior 

We get notified of a completed PATCH request

### LINKED ISSUE #3032: Front-end work for "locking" library templates
## Description

This issue tracks the front-end work needed to implement #2991 (back-end part is #3005).

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

This includes the huge #3120 PR and we should wait for it to be merged before tackling this.

## Related issues

Fixes #3032 
Fixes #2994

### CODE REVIEW

File: jsapp/xlform/src/view.rowDetail.coffee
Comment (duvld) [inline jsapp/xlform/src/view.rowDetail.coffee:31]: class is not referenced anywhere else, maybe leftover from previous changes?

**Review by duvld — COMMENTED** [review]
Small questions and tiny changes. There's some grammar mistakes in exposed UI strings that I'll commit myself, but beyond that it's good to go :)

File: jsapp/js/components/locking/formLockedMessage.es6
Comment (duvld) [reply jsapp/js/components/locking/formLockedMessage.es6]: https://github.com/kobotoolbox/kpi/pull/3127/commits/c9649d61e10f794f1d334178ec4d0b332ab180c1

File: jsapp/js/components/locking/lockingUtils.es6
Comment (magicznyleszek) [reply jsapp/js/components/locking/lockingUtils.es6:23]: Added a comment next to `QUESTION_TYPES` :)

File: jsapp/js/components/formSubScreens.es6
Comment (magicznyleszek) [reply jsapp/js/components/formSubScreens.es6:34]: I think this was something for testing or fixing performance isssues. The key element is `this.props.randdelay`, which was never used now.

File: jsapp/js/bem.es6
Comment (magicznyleszek) [reply jsapp/js/bem.es6:55]: Right! I replaced it with `.locked-features` later and forgot to remove it

Comment (magicznyleszek) [reply jsapp/js/bem.es6:55]: Removed it now :)

File: jsapp/js/components/locking/lockingUtils.es6
Comment (magicznyleszek) [reply jsapp/js/components/locking/lockingUtils.es6:201]: This would not break anything, just the labels in few places will not say "fully locked".

Comment (magicznyleszek) [reply jsapp/js/components/locking/lockingUtils.es6:201]: But checking it will not be easy. `lock_all` in the code means that form, all questions and all groups have restrictions from the list `LOCK_ALL_RESTRICTION_NAMES` (currently all known restrictions)

Comment (magicznyleszek) [reply jsapp/js/components/locking/lockingUtils.es6:201]: I've added a comment #fixed :P

File: jsapp/xlform/src/view.choices.templates.coffee
Comment (magicznyleszek) [reply jsapp/xlform/src/view.choices.templates.coffee:4]: `js-` classes are a widely accepted way of naming a selector that is only for JS code and should not have any stylings on it. This is made so, because if you have a class name with some styles, e.g. `card__label` and it both has some styles and is being used in JS, and some UI update comes and styles are dropped, you could drop `card__label` without noticing that you broke JS.

File: jsapp/xlform/src/view.row.coffee
Comment (magicznyleszek) [reply jsapp/xlform/src/view.row.coffee:295]: It is! Removed :)

File: jsapp/xlform/src/view.rowDetail.coffee
Comment (magicznyleszek) [reply jsapp/xlform/src/view.rowDetail.coffee:31]: Ah right, another leftover. The manage translations restriction changed so many times :P

File: jsapp/xlform/src/view.row.coffee
Comment (magicznyleszek) [reply jsapp/xlform/src/view.row.coffee:330]: Missed this for groups, fixed :)

File: jsapp/xlform/src/view.choices.templates.coffee
Comment (duvld) [reply jsapp/xlform/src/view.choices.templates.coffee:4]: that makes a lot of sense, it sounds pretty much mandatory with working with jquery

**Review by duvld — APPROVED** [review]

File: kpi/models/asset.py
Comment (jnm) [inline kpi/models/asset.py:27]: typo :)

File: kpi/models/asset.py
Comment (jnm) [inline kpi/models/asset.py:246]: typo :)

File: kpi/models/asset.py
Comment (jnm) [inline kpi/models/asset.py:250]: i wouldn't follow the example of `_remove_empty_expressions()` here: these methods, which are really static methods, don't seem to add anything. unless i'm missing something, directly call the functions that were imported from formpack.

File: kpi/models/import_export_task.py
Comment (jnm) [inline kpi/models/import_export_task.py:764]: kind of pedantic, but in the interest of clarity: `b64decode()` returns a `bytes`, not a `str`

**Review by jnm — CHANGES_REQUESTED** [review]
@joshuaberetta, just a few small things. i opted to add my comments here instead of trying to reopen #3082.

maybe we can wait until kobotoolbox/formpack#238 is merged, then add a commit to this branch pointing at the latest formpack, and then merge this PR.

File: kpi/models/asset.py
Comment (joshuaberetta) [reply kpi/models/asset.py:250]: I just followed the structure that existed in the class, but I'll change it to something more sensible 😄

**Review by jnm — APPROVED** [review]
:+1:

**Pull request merged**
